gvr-3dcursor/app/src/main/java/org/gearvrf/io/cursor/CursorScript.java

```diff
@@ -920,7 +931,10 @@ void handleClickReleased(CursorEvent event) {
             }
 
             targetCursor.setCursorTheme(crystalTheme);
-            targetCursor.setPosition(cursorPosition.x, cursorPosition.y, cursorPosition.z);
+            if(targetCursor.getIoDevice()!= null && !targetCursor.getIoDevice().getDeviceId()
+                    .equals(GEARVR_DEVICE_ID)) {
+                targetCursor.setPosition(cursorPosition.x, cursorPosition.y, cursorPosition.z);
```

Comment:
Cleaner to create a private function to set position. You could add a check for GEARVR_DEVICE_ID whenever that method is invoked.